contracts/0.8.25/vaults/VaultHub.sol

```diff
@@ -998,6 +1138,181 @@ contract VaultHub is PausableUntilWithRoles {
         emit VaultInOutDeltaUpdated(_vault, inOutDelta_.value);
     }
 
+    /**
+     * @notice Updates the unsettled Lido fees obligations based on the report cumulative Lido fees
+     * @param _vault The address of the vault
+     * @param _reportCumulativeLidoFees The cumulative Lido fees reported in the report
+     */
+    function _checkAndUpdateLidoFeesObligations(
+        address _vault,
+        VaultObligations storage _obligations,
+        uint256 _reportCumulativeLidoFees
+    ) internal {
+        uint256 cumulativeSettledLidoFees = _obligations.settledLidoFees;
+        uint256 cumulativeLidoFees = cumulativeSettledLidoFees + _obligations.unsettledLidoFees;
+        if (_reportCumulativeLidoFees < cumulativeLidoFees) {
+            revert InvalidFees(_vault, _reportCumulativeLidoFees, cumulativeLidoFees);
+        }
+
+        // update unsettled lido fees
+        uint256 unsettledLidoFees = _reportCumulativeLidoFees - cumulativeSettledLidoFees;
+        if (unsettledLidoFees != _obligations.unsettledLidoFees) {
+            _obligations.unsettledLidoFees = uint128(unsettledLidoFees);
+            emit LidoFeesUpdated(_vault, unsettledLidoFees, cumulativeSettledLidoFees);
+        }
+    }
+
+    /**
+     * @notice Calculates a settlement plan based on vault balance and obligations
+     * @param _vault The address of the vault
+     * @param _record The record of the vault
+     * @param _obligations The obligations of the vault to be settled
+     * @return valueToRebalance The ETH amount to be rebalanced for redemptions
+     * @return sharesToRebalance The shares to be rebalanced for redemptions
+     * @return valueToTransferToLido The ETH amount to be sent to the Lido
+     * @return unsettledRedemptions The remaining redemptions after the planned settlement
+     * @return unsettledLidoFees The remaining Lido fees after the planned settlement
+     * @return totalUnsettled The total ETH value of obligations remaining after the planned settlement
+     */
+    function _planSettlement(
+        address _vault,
+        VaultRecord storage _record,
+        VaultObligations storage _obligations
+    ) internal view returns (
+        uint256 valueToRebalance,
+        uint256 sharesToRebalance,
+        uint256 valueToTransferToLido,
+        uint256 unsettledRedemptions,
+        uint256 unsettledLidoFees,
+        uint256 totalUnsettled
+    ) {
+        uint256 vaultBalance = _vault.balance;
+        uint256 cappedRedemptionsShares = Math256.min(_record.liabilityShares, _getSharesByPooledEthRoundUp(_obligations.redemptions));
```

Comment:
let's redo this:
1) s = getSharesByPooledEth
2) x = getPooledEtherBySharesRoundUp
3) if x < redemptions, then s++; // ??? do we need this

Reply:
share rate between 2 and 4 to test it out